### PR TITLE
SPAdes: refactor

### PR DESCRIPTION
**Pull request by quantenzitrone**
## Description of changes

renamed to fit package naming conventions
split of https://github.com/NixOS/nixpkgs/pull/319219

I also pointed `src` and `meta.homepage` to the new website and github repo, since the old one is dead

there is a new version as well, but I didn't want to touch this as much 
@bzizou you might want to do that

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sp/spades/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  buildInputs = [
+    zlib
+    bzip2
+    python3
+  ];
+
+  doCheck = true;
+
+  sourceRoot = "SPAdes-${finalAttrs.version}/src";
```
Comment (superherointj) [inline pkgs/by-name/sp/spades/package.nix:29]: Please move `sourceRoot` just after `src`.

File: pkgs/by-name/sp/spades/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  env.CXXFLAGS = toString [
```
Comment (superherointj) [inline pkgs/by-name/sp/spades/package.nix:31]: `env` is configure phase. Follow phases order:

https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

(Move it to before build phase)

File: pkgs/by-name/sp/spades/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  zlib,
+  bzip2,
+  cmake,
+  python3,
+}:
+stdenv.mkDerivation (finalAttrs: {
```
Comment (superherointj) [inline pkgs/by-name/sp/spades/package.nix:10]:
```suggestion
}:

stdenv.mkDerivation (finalAttrs: {
```

(Just a line break)

File: pkgs/by-name/sp/spades/package.nix (hunk 2 above)
Comment (quantenzitrone) [reply pkgs/by-name/sp/spades/package.nix:31]: i can't find env in the docs :skull:
nix* docs as good as always

Comment (superherointj) [reply pkgs/by-name/sp/spades/package.nix]: `env` being environment variable. `CXXFLAGS` being configure phase. This is part of 'configuration' of build.
Just move it to configure phase.

**Review by superherointj — APPROVED** [review]
Code-wise looks good.
I'm waiting for CI to merge w/ squash (to fix commit message).

**Comment by bzizou** [comment]
Thank you for that! I'll check for the 4.0.0 version...

**Pull request merged**